connection/src/test/java/com/netcracker/ConnectionTest.java

```diff
@@ -0,0 +1,42 @@
+package com.netcracker;
+
+import com.netcracker.domain.Message;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.net.SocketAddress;
+
+import static com.netcracker.domain.MessageType.TEXT;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+public class ConnectionTest {
```

Comment:
The main reason is that I cannot correctly instantiate or mock InputStream in order to use it in Connection's constructor. I got CorruptedStreamException or infinite instatiating. I mean use it like "when(mockedSocket.getInputStream).thenReturn(mockedInputStream)". Maybe I can test it better when I will write tests for server or client class, however it will be integration tests then, because I won't mock Connection.